src/t_plist.c

```diff
@@ -338,11 +338,7 @@ int pmLsetCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
   }
   size_t val_len;
   const char *val = RedisModule_StringPtrLen(argv[3], &val_len);
-  if (idx >= 0) {
-    KVDKListInsertAfter(engine, iter, val, val_len);
-  } else {
-    KVDKListInsertBefore(engine, iter, val, val_len);
-  }
+  s = KVDKListReplace(engine, iter, val, val_len);
```

Comment:
can we confirm that the result of pm.lset is the same of the original behavior of redis after the modification?
E.g.
redis:6379> RPUSH mylist "one"
(integer) 1
redis:6379> RPUSH mylist "two"
(integer) 2
redis:6379> RPUSH mylist "three"
(integer) 3
redis:6379> LSET mylist 0 "four"
"OK"
redis:6379> LSET mylist -2 "five"
"OK"
redis:6379> LRANGE mylist 0 -1
1) "four"
2) "five"
3) "three"

Reply:
Yes, this is confirmed.

Reply:
Ok, good